src/sentry/search/events/builder/discover.py

```diff
@@ -583,13 +583,17 @@ def resolve_params(self) -> list[WhereType]:
         if self.end:
             conditions.append(Condition(self.column("timestamp"), Op.LT, self.end))
 
-        conditions.append(
-            Condition(
-                self.column("project_id"),
-                Op.IN,
-                self.params.project_ids,
+        # The clause will prevent calling Snuba with an empty list of projects, thus, returning
+        # no data. It will not instead complain with:
+        # sentry.utils.snuba.UnqualifiedQueryError: validation failed for entity events: missing required conditions for project_id
```

Comment:
🤔 Wouldn't it be better to raise an error here instead of calling snuba? That just means extra processing when we already know no project_ids isn't a legitimate query

Reply:
The convo got lost from the previous PR.

I started doing so in #69575 but it started getting to be a lot of tests that need fixing. I want to get this merged since it helps anyone doing local development.

Reply:
🤔 I'd honestly prefer we just check if we're running tests and only raise outside of that if updating the tests is too much work